### ISSUE TITLE
Not support HTMLVideoElement ?

### ISSUE THREAD

**Issue by preflower** [lekoala/barcode-detector-zbar#6]
i try to pass `video` element to `detector.detect` api, but it return `No width or height` reject

**Comment by lekoala** [comment]
does it has a width and height ?
did you try the sample code in the demo to see if it works for you ?

**Comment by preflower** [comment]
video need `videoHeight`, `videoWidth` to get height, width; and if u wanna get element width/height, need use `offsetWidth`, `offsetHeight`

**Comment by lekoala** [comment]
i don't know, all i can say is that it works for me :) if you have a codepen or something or make a PR you are welcome to do so

**Comment by preflower** [comment]
https://codepen.io/preflower/pen/ExoQOOK

**Comment by preflower** [comment]
> i don't know, all i can say is that it works for me :) if you have a codepen or something or make a PR you are welcome to do so

https://codepen.io/preflower/pen/ExoQOOK

**Comment by preflower** [comment]
![image](https://user-images.githubusercontent.com/30513719/161940119-1226f9ab-1a94-4296-a791-7593f9e685fd.png)

**Comment by lekoala** [comment]
yes i see. but nothing is preventing you to set an actual width on the video element before detecting? not ideal, but it should work fine

**Comment by preflower** [comment]
video need use `videoWidth/Height` to get actual width/height, it has nothing to do with `ideal`,  even if i set `video.width/height`, is element w/h, not video actual w/h...
